packages/esm-service-queues-app/src/queue-services/queue-service-form.workspace.tsx

```diff
@@ -18,85 +22,84 @@ import { saveQueue, useServiceConcepts } from './queue-service.resource';
 import { useQueueLocations } from '../create-queue-entry/hooks/useQueueLocations';
 import styles from './queue-service-form.scss';
 
+const QueueServiceSchema = z.object({
+  queueName: z.string().min(1, { message: 'Queue name is required' }),
+  queueConcept: z.string().min(1, { message: 'Queue concept is required' }),
+  userLocation: z.string().min(1, { message: 'Queue location is required' }),
+});
+type QueueServiceFormData = z.infer<typeof QueueServiceSchema>;
+
 const QueueServiceForm: React.FC<DefaultWorkspaceProps> = ({ closeWorkspace }) => {
   const { t } = useTranslation();
   const { queueConcepts } = useServiceConcepts();
   const { queueLocations } = useQueueLocations();
-  const [queueName, setQueueName] = useState('');
-  const [queueConcept, setQueueConcept] = useState('');
-  const [isMissingName, setMissingName] = useState(false);
-  const [isMissingQueue, setMissingQueue] = useState(false);
-  const [isMissingLocation, setMissingLocation] = useState(false);
-  const [userLocation, setUserLocation] = useState('');
-
-  const createQueue = useCallback(
-    (event) => {
-      event.preventDefault();
-
-      if (!queueName) {
-        setMissingName(true);
-        return;
-      }
-      if (!queueConcept) {
-        setMissingQueue(true);
-        return;
-      }
-      if (!userLocation) {
-        setMissingLocation(true);
-        return;
-      }
 
-      setMissingName(false);
-      setMissingQueue(false);
-      setMissingLocation(false);
+  const {
+    control,
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<QueueServiceFormData>({
+    resolver: zodResolver(QueueServiceSchema),
+    defaultValues: {
+      queueName: '',
+      queueConcept: '',
+      userLocation: '',
+    },
+  });
 
-      saveQueue(queueName, queueConcept, queueName, userLocation).then(
-        ({ status }) => {
-          if (status === 201) {
-            showSnackbar({
-              title: t('addQueue', 'Add queue'),
-              kind: 'success',
-              subtitle: t('queueAddedSuccessfully', 'Queue added successfully'),
-            });
-            closeWorkspace();
-            mutate(`${restBaseUrl}/queue?${userLocation}`);
-            mutate(`${restBaseUrl}/queue?location=${userLocation}`);
-          }
-        },
-        (error) => {
+  const createQueue = (data: QueueServiceFormData) => {
+    saveQueue(data.queueName, data.queueConcept, data.queueName, data.userLocation).then(
+      ({ status }) => {
```

Comment:
`.then` and showing the snackbar for success a `.catch` to catch the error afterwards should be sufficient, rather than checking for the status and doing an error inside the `.then`